src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+)
+from ..irf.benchmarks import (
+    AngularResolutionMakerBase,
+    EnergyBiasResolutionMakerBase,
+    SensitivityMakerBase,
+)
+from ..irf.irfs import (
+    BackgroundRateMakerBase,
+    EffectiveAreaMakerBase,
+    EnergyMigrationMakerBase,
+    PsfMakerBase,
+)
+
+
+class IrfTool(Tool):
+    name = "ctapipe-make-irf"
+    description = "Tool to create IRF files in GAD format"
+
+    do_background = Bool(
+        True,
+        help="Compute background rate IRF using supplied files",
+    ).tag(config=True)
+
+    do_benchmarks = Bool(
+        False,
+        help="Produce IRF related benchmarks",
+    ).tag(config=True)
+
+    range_check_error = Bool(
+        False,
+        help="Raise error if asking for IRFs outside range where cut optimisation is valid",
+    ).tag(config=True)
+
+    cuts_file = traits.Path(
+        default_value=None, directory_ok=False, help="Path to optimized cuts input file"
+    ).tag(config=True)
+
+    gamma_file = traits.Path(
+        default_value=None, directory_ok=False, help="Gamma input filename and path"
+    ).tag(config=True)
+
+    gamma_target_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.CRAB_HEGRA,
+        help="Name of the spectra used for the simulated gamma spectrum",
+    ).tag(config=True)
+
+    proton_file = traits.Path(
+        default_value=None,
+        allow_none=True,
+        directory_ok=False,
+        help="Proton input filename and path",
+    ).tag(config=True)
+
+    proton_target_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.IRFDOC_PROTON_SPECTRUM,
+        help="Name of the spectra used for the simulated proton spectrum",
+    ).tag(config=True)
+
+    electron_file = traits.Path(
+        default_value=None,
+        allow_none=True,
+        directory_ok=False,
+        help="Electron input filename and path",
+    ).tag(config=True)
+
+    electron_target_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.IRFDOC_ELECTRON_SPECTRUM,
+        help="Name of the spectra used for the simulated electron spectrum",
+    ).tag(config=True)
+
+    chunk_size = Integer(
+        default_value=100000,
+        allow_none=True,
+        help="How many subarray events to load at once while selecting.",
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        default_value="./IRF.fits.gz",
+        allow_none=False,
+        directory_ok=False,
+        help="Output file",
+    ).tag(config=True)
+
+    obs_time = AstroQuantity(
+        default_value=u.Quantity(50, u.hour),
+        physical_type=u.physical.time,
+        help="Observation time in the form ``<value> <unit>``",
+    ).tag(config=True)
+
+    edisp_parameterization = traits.ComponentName(
+        EnergyMigrationMakerBase,
+        default_value="EnergyMigration2dMaker",
+        help="The parameterization of the energy migration to be used.",
+    ).tag(config=True)
+
+    aeff_parameterization = traits.ComponentName(
```

Comment:
It's a bit weird to have these option be called "<component>_parametrization" but then the class names all end on "Maker".

Can we make this consistent? Either call the options `..._maker` or the classes `...Parametrization`?